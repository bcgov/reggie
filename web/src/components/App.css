.container {
  width: 80%;
  max-width: 1200px;
  margin: 3em auto;
  padding: 1em;
}

.container * {
  box-sizing: border-box;
}

p {
  color: #003366;
}

a {
  color: #003366;
  text-decoration: underline;
}

a:hover {
  color: #003366;
}

button {
  font-size: 12px;
}

.btn.btn-primary {
  background-color: #003366; /* Green */
  border: none;
  border-radius: 5px;
  color: white;
  padding: 10px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  cursor: pointer;
}

.btn.btn-primary:hover {
text-decoration: underline;
opacity: 0.9;
}

div {
  text-align: center;
}

h1 {
  margin-bottom: 50px;
}

.container h1 {
  margin-bottom: 20px;
  border-bottom: 1px solid #e5e5e5;
}

.display-linebreak {
  white-space: pre-line;
}

.flex-container {
  display: flex;
  justify-content: center;
}

.flex-container > div {
  margin: 0 10px;
}

@media screen and (max-width: 500px) {
  .container h1 {
    font-size: 1.5rem;
  }

  .btn.btn-primary {
    font-size: 13px;
    padding: 10px;
  }
}