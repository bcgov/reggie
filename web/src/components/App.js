//
// Reggie Web
//
// Copyright Â© 2018 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Created by Shelly Xue Han on 2019-01-10.
//

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Route, Switch } from 'react-router-dom';
import { authenticateFailed, authenticateSuccess } from '../actions';
import { authorize } from '../actionCreators';
import implicitAuthManager from '../auth';
import { AUTH_CODE, SELF_SERVER_APP } from '../constants';
import { Home, RocketChat, Rejection, Email, Verify } from '../containers';
import Layout from '../hoc/Layout';
import PrivateRoute from './Navigation/PrivateRoute';
import { LoginRoute } from './Navigation/LoginRoute';
import './App.css';
import AuthModal from './Auth/AuthModal';

export class App extends Component {
  componentDidMount = () => {
    implicitAuthManager.registerHooks({
      onAuthenticateSuccess: () => this.props.login(),
      onAuthenticateFail: () => this.props.logout(),
      // onAuthLocalStorageCleared: () => this.props.logout(),
    });
    // don't call function if on localhost
    if (!window.location.host.match(/localhost/)) {
      implicitAuthManager.handleOnPageLoad();
    }
    try {
      const iamId = implicitAuthManager.idToken.data.sub;
      // if user authenticated, try authorization:
      if (iamId) this.props.authorize(SELF_SERVER_APP.ROCKETCHAT.NAME, iamId);
    } catch (err) {
      console.log('---implicitAuthManager----not logged in');
    }
  };

  render() {
    // Get the current authorization status code for private route:
    const currAuthCode = this.props.authorization.authCode;
    return (
      <Layout>
        <AuthModal isAuthenticated={this.props.authentication.isAuthenticated} />
        <Switch>
          <PrivateRoute
            path="/rocketChat"
            authorization={this.props.authorization}
            component={RocketChat}
            shouldRender={currAuthCode === AUTH_CODE.AUTHORIZED}
            redirectTo="/"
          />
          <Route path="/rejection" component={Rejection} />
          <Route path="/email" component={Email} />
          <Route
            path="/verify"
            component={Verify}
            authentication={this.props.authentication}
            authorization={this.props.authorization}
            verifyEmail={this.props.verifyEmail}
          />
          <Route
            path="/ssoLogin/:loginIdp"
            component={({ match }) => {
              if (match.params.loginIdp) {
                implicitAuthManager.config.kcIDPHint = match.params.loginIdp;
              }
              window.location = implicitAuthManager.getSSOLoginURI();
            }}
          />
          <Route path="/login" component={LoginRoute} />
          <Route
            path="/logout"
            component={() => {
              this.props.logout();
              window.location = implicitAuthManager.getSSOLogoutURI();
            }}
          />
          <Route path="/" component={Home} authentication={this.props.authentication} />
        </Switch>
      </Layout>
    );
  }
}

function mapStateToProps(state) {
  return {
    authentication: state.authentication,
    authorization: state.authorization,
    verifyEmail: state.verifyEmail,
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      login: () => dispatch(authenticateSuccess()),
      logout: () => dispatch(authenticateFailed()),
      authorize,
    },
    dispatch
  );
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
