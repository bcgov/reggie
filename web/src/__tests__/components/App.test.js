import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../../components/App';

describe('App Component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <App
        authentication={{ isAuthenticated: true }}
        authorization={{ isAuthorizing: true, authCode: 'new' }}
        verifyEmail={{ verfied: false }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
