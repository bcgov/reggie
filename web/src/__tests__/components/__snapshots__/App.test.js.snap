// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    authentication={
      Object {
        "isAuthenticated": true,
      }
    }
    authorization={
      Object {
        "authCode": "new",
        "isAuthorizing": true,
      }
    }
    verifyEmail={
      Object {
        "verfied": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <AuthModal
          isAuthenticated={true}
        />,
        <Switch>
          <PrivateRoute
            authorization={
              Object {
                "authCode": "new",
                "isAuthorizing": true,
              }
            }
            component={[Function]}
            path="/rocketChat"
            redirectTo="/"
            shouldRender={false}
          />
          <Route
            component={[Function]}
            path="/rejection"
          />
          <Route
            component={[Function]}
            path="/email"
          />
          <Route
            authentication={
              Object {
                "isAuthenticated": true,
              }
            }
            authorization={
              Object {
                "authCode": "new",
                "isAuthorizing": true,
              }
            }
            component={[Function]}
            path="/verify"
            verifyEmail={
              Object {
                "verfied": false,
              }
            }
          />
          <Route
            component={[Function]}
            path="/ssoLogin/:loginIdp"
          />
          <Route
            component={[Function]}
            path="/login"
          />
          <Route
            component={[Function]}
            path="/logout"
          />
          <Route
            authentication={
              Object {
                "isAuthenticated": true,
              }
            }
            component={[Function]}
            path="/"
          />
        </Switch>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "isAuthenticated": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <PrivateRoute
              authorization={
                Object {
                  "authCode": "new",
                  "isAuthorizing": true,
                }
              }
              component={[Function]}
              path="/rocketChat"
              redirectTo="/"
              shouldRender={false}
            />,
            <Route
              component={[Function]}
              path="/rejection"
            />,
            <Route
              component={[Function]}
              path="/email"
            />,
            <Route
              authentication={
                Object {
                  "isAuthenticated": true,
                }
              }
              authorization={
                Object {
                  "authCode": "new",
                  "isAuthorizing": true,
                }
              }
              component={[Function]}
              path="/verify"
              verifyEmail={
                Object {
                  "verfied": false,
                }
              }
            />,
            <Route
              component={[Function]}
              path="/ssoLogin/:loginIdp"
            />,
            <Route
              component={[Function]}
              path="/login"
            />,
            <Route
              component={[Function]}
              path="/logout"
            />,
            <Route
              authentication={
                Object {
                  "isAuthenticated": true,
                }
              }
              component={[Function]}
              path="/"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "authorization": Object {
                "authCode": "new",
                "isAuthorizing": true,
              },
              "component": [Function],
              "path": "/rocketChat",
              "redirectTo": "/",
              "shouldRender": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/rejection",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/email",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "authentication": Object {
                "isAuthenticated": true,
              },
              "authorization": Object {
                "authCode": "new",
                "isAuthorizing": true,
              },
              "component": [Function],
              "path": "/verify",
              "verifyEmail": Object {
                "verfied": false,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/ssoLogin/:loginIdp",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/logout",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "authentication": Object {
                "isAuthenticated": true,
              },
              "component": [Function],
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <AuthModal
            isAuthenticated={true}
          />,
          <Switch>
            <PrivateRoute
              authorization={
                Object {
                  "authCode": "new",
                  "isAuthorizing": true,
                }
              }
              component={[Function]}
              path="/rocketChat"
              redirectTo="/"
              shouldRender={false}
            />
            <Route
              component={[Function]}
              path="/rejection"
            />
            <Route
              component={[Function]}
              path="/email"
            />
            <Route
              authentication={
                Object {
                  "isAuthenticated": true,
                }
              }
              authorization={
                Object {
                  "authCode": "new",
                  "isAuthorizing": true,
                }
              }
              component={[Function]}
              path="/verify"
              verifyEmail={
                Object {
                  "verfied": false,
                }
              }
            />
            <Route
              component={[Function]}
              path="/ssoLogin/:loginIdp"
            />
            <Route
              component={[Function]}
              path="/login"
            />
            <Route
              component={[Function]}
              path="/logout"
            />
            <Route
              authentication={
                Object {
                  "isAuthenticated": true,
                }
              }
              component={[Function]}
              path="/"
            />
          </Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isAuthenticated": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <PrivateRoute
                authorization={
                  Object {
                    "authCode": "new",
                    "isAuthorizing": true,
                  }
                }
                component={[Function]}
                path="/rocketChat"
                redirectTo="/"
                shouldRender={false}
              />,
              <Route
                component={[Function]}
                path="/rejection"
              />,
              <Route
                component={[Function]}
                path="/email"
              />,
              <Route
                authentication={
                  Object {
                    "isAuthenticated": true,
                  }
                }
                authorization={
                  Object {
                    "authCode": "new",
                    "isAuthorizing": true,
                  }
                }
                component={[Function]}
                path="/verify"
                verifyEmail={
                  Object {
                    "verfied": false,
                  }
                }
              />,
              <Route
                component={[Function]}
                path="/ssoLogin/:loginIdp"
              />,
              <Route
                component={[Function]}
                path="/login"
              />,
              <Route
                component={[Function]}
                path="/logout"
              />,
              <Route
                authentication={
                  Object {
                    "isAuthenticated": true,
                  }
                }
                component={[Function]}
                path="/"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "authorization": Object {
                  "authCode": "new",
                  "isAuthorizing": true,
                },
                "component": [Function],
                "path": "/rocketChat",
                "redirectTo": "/",
                "shouldRender": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/rejection",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/email",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "authentication": Object {
                  "isAuthenticated": true,
                },
                "authorization": Object {
                  "authCode": "new",
                  "isAuthorizing": true,
                },
                "component": [Function],
                "path": "/verify",
                "verifyEmail": Object {
                  "verfied": false,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/ssoLogin/:loginIdp",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/logout",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "authentication": Object {
                  "isAuthenticated": true,
                },
                "component": [Function],
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
